src/components/RemountOnThemeChange.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useRef, useState } from "react"
+import { useAtom } from "jotai"
+
+import { $theme } from "../store/theme"
+
+interface Props {
+  children: React.ReactNode
+}
+
+/**
+ * WORKAROUND: remount the children when the theme changes.
+ *
+ * This causes the charts data to be re-fetched and re-rendered
+ * when the theme changes.
+ */
+export function RemountOnThemeChange(props: Props) {
```

Comment:
This seems too overzealous in my opinion. I think we could achieve the same effect by forcing `key`.

```ts
export function RemountOnThemeChange(props: Props) {
  const [themeName] = useAtom($theme)
  return <Fragment key={themeName}>{props.children}</Fragment>
}
```

Could you see if this solves all your problems, I tried this with the dashboard page `admin/analytics/:id` and it reloaded the dashboard when changing themes